bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/HomekitSecuritySystemImpl.java

```diff
@@ -43,123 +47,76 @@
  */
 public class HomekitSecuritySystemImpl extends AbstractHomekitAccessoryImpl implements SecuritySystemAccessory {
     private final Logger logger = LoggerFactory.getLogger(HomekitSecuritySystemImpl.class);
+    private final Map<CurrentSecuritySystemStateEnum, String> currentStateMapping = new EnumMap(
```

Comment:
```suggestion
    private final Map<CurrentSecuritySystemStateEnum, String> currentStateMapping = new EnumMap<>(
```

Reply:
i get 
_`'<>' cannot be used with anonymous classes`_

should keep it as it is or change to
`
    private final Map<TargetSecuritySystemStateEnum, String> targetStateMapping = new EnumMap<TargetSecuritySystemStateEnum, String>(`
?

Reply:
Honestly, creating an anonymous subclass of EnumMap for the sole purpose of populating initial values is something I frown upon because it mixes up class implementation with class usage which should generally be two separate concerns when coding. The error you are seeing is one such side-effect of that mixing up.

But in order to not force even more changes right now I suggest to just change it to `private final Map<TargetSecuritySystemStateEnum, String> targetStateMapping = new EnumMap<TargetSecuritySystemStateEnum, String>(`

Reply:
i will change as you suggested. 

but how you would do it without anonymous subclass? with a dedicate classes, e.g. SecurityCurrentStateMapping? or something else? originally i was thinking about kind of xml based configuration, it was too complex and too error-prone at the end.

Reply:
You would typically populate the map in the HomeSecuritySystemImpl constructor, or you could (although I don't really recommend it either since it is kinda unorthodox) just put the population logic in a raw statement block `{...}` right below your field declaration. It is like static initialization but you just remove the static keyword.

Reply:
ok.  understood. i will keep it as it is for now. we consider to change it all places at once to constructor based init in a later PR